torchdata/dataloader2/reading_service.py

```diff
@@ -146,6 +146,73 @@ def __new__(cls, *args, **kwargs):
         return MultiProcessingReadingService(*args, **kwargs)
 
 
+class SingleProcessingReadingService(ReadingServiceInterface):
+    r"""
+    Default ReadingService to serve the ``DataPipe` graph in the main process,
+    and apply graph settings like determinism control to the graph.
+
+    Args:
+        init_fn: (Callable, optional): Custom function to be called when the main
+            process starts to iterate over ``DataPipe`` graph.
+        reset_fn: (Callable, optional): Custom function to be called at the beginning
+            of each epoch with ``DataPipe``, ``WorkerInfo`` and ``SeedGenerator``
+            as the expected arguments.
+    """
+    _init_fn: Optional[Callable[[DataPipe, WorkerInfo], DataPipe]]
+    _reset_fn: Optional[Callable[[DataPipe, WorkerInfo, SeedGenerator], DataPipe]]
+    _end_datapipe: Optional[DataPipe]
+
+    def __init__(
+        self,
+        init_fn: Optional[Callable[[DataPipe, WorkerInfo], DataPipe]] = None,
+        reset_fn: Optional[Callable[[DataPipe, WorkerInfo, SeedGenerator], DataPipe]] = None,
```

Comment:
Maybe adding one is better. But, it will be BC-breaking as previously default `DataLoader2` won't do any prefetching. So, I guess I will keep that a separate small PR.

Reply:
Maybe we can add the option but leave the default as no prefetch?